demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -470,24 +494,34 @@ static void prvAzureDemoTask( void * pvParameters )
         }
 
         xResult = AzureIoTHubClient_UnsubscribeProperties( &xAzureIoTHubClient );
-        configASSERT( xResult == eAzureIoTSuccess );
 
-        xResult = AzureIoTHubClient_UnsubscribeCommand( &xAzureIoTHubClient );
-        configASSERT( xResult == eAzureIoTSuccess );
+        /* if there are network issues, this will fail and we should still close the network connection and restart */
+        if( xResult == eAzureIoTSuccess )
+        {
+            xResult = AzureIoTHubClient_UnsubscribeCommand( &xAzureIoTHubClient );
+            configASSERT( xResult == eAzureIoTSuccess );
 
-        /* Send an MQTT Disconnect packet over the already connected TLS over
-         * TCP connection. There is no corresponding response for the disconnect
-         * packet. After sending disconnect, client must close the network
-         * connection. */
-        xResult = AzureIoTHubClient_Disconnect( &xAzureIoTHubClient );
-        configASSERT( xResult == eAzureIoTSuccess );
+            /* Send an MQTT Disconnect packet over the already connected TLS over
+             * TCP connection. There is no corresponding response for the disconnect
+             * packet. After sending disconnect, client must close the network
+             * connection. */
+            xResult = AzureIoTHubClient_Disconnect( &xAzureIoTHubClient );
+            configASSERT( xResult == eAzureIoTSuccess );
```

Comment:
Assets should only be used on precondition that must be true. In this case xRestult can be a valid failure case. These should be just logging items. Example, what happens if you build release, these checks are not there anymore.

Reply:
I agree - I think we probably need to take a look at most of the asserts in the samples (~300) and see how many have valid failure cases that should have defined ways of handling them/logging and continuing. During standup, we had decided on the quick fix for this issue of logging on network issues and restarting the sample, but I can create a separate work item to take a look at the asserts in the samples and refactor them. Unfortunately for the ESP32, the assert function is in the esp library, so we can't easily override it like with some of the other devices.